rust/tw_hd_wallet/src/ton/mnemonic.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+use crate::bip39::bip39_english::BIP39_WORDS_MAP;
+use crate::{WalletError, WalletResult};
+use zeroize::ZeroizeOnDrop;
+
+#[derive(Debug, ZeroizeOnDrop)]
+pub struct TonMnemonic(String);
+
+impl TonMnemonic {
+    pub const WORDS_LEN: usize = 24;
+
+    /// Creates `TonMnemonic` from a 24-words string.
+    ///
+    /// Please note there can be a passphrase required to be used to derive a key-pair.
+    /// See [`TonWallet::new`].
+    pub fn new(mnemonic: &str) -> WalletResult<TonMnemonic> {
```

Comment:
It seems that the validation in this function is very basic. It is recommended to add an `is_basic_seed` check.

Reply:
`is_basic_seed` requires the pair of mnemonic and passphrase to generate the entropy. It's an overhead for `TonMnemonic` to also validate the passphrase. However, the idea of this structure is to validate the number of words and the words themselves (if they're from BIP39 dictionary)

Reply:
1. `is_basic_seed` only requires the parameter `entropy`.
2. The function signature is TonMnemonic::new(mnemonic). From the function name alone, it's not clear that it only checks the length and words, and does not verify other validity aspects. If you don't check everything, it can cause significant mental strain for someone reviewing this code. I suggest you refactor it.

Reply:
Thank you for the explanation, I've replaced `TonMnemonic` structure with a single `validate_mnemonic_words` function